src/rpc/mission.rs

```diff
@@ -81,13 +78,25 @@ impl MissionService for MissionRpc {
             .await?
             .into_inner();
 
-        let dt = Utc
-            .ymd(current.year, current.month, current.day)
-            .and_hms(0, 0, 0);
-        let dt = dt + Duration::seconds(current.time as i64);
-
         Ok(Response::new(mission::GetScenarioCurrentTimeResponse {
-            datetime: dt.format("%Y-%m-%dT%H:%M:%SZ").to_string(),
+            datetime: to_datetime(current.year, current.month, current.day, current.time)?,
         }))
     }
 }
+
+fn to_datetime(year: i32, month: u32, day: u32, time: f64) -> Result<String, Status> {
+    let month = u8::try_from(month)
+        .map_err(|err| Status::internal(format!("received invalid month: {}", err)))?;
+    let month = Month::try_from(month)
+        .map_err(|err| Status::internal(format!("received invalid month: {}", err)))?;
+    let day = u8::try_from(day)
+        .map_err(|err| Status::internal(format!("received invalid day: {}", err)))?;
+    let date = Date::from_calendar_date(year, month, day)
+        .map_err(|err| Status::internal(format!("received invalid date: {}", err)))?;
+    let time = Time::from_hms(0, 0, 0).unwrap() + Duration::seconds(time as i64);
+    let datetime = PrimitiveDateTime::new(date, time).assume_offset(UtcOffset::UTC);
+
+    datetime.format(&Rfc3339).map_err(|err| {
+        Status::internal(format!("failed to format date as ISO 8601 string: {}", err))
+    })
+}
```

Comment:
`igrf` is based on the `time` crate, which is why I've migrated the code above to the `time` crated too, so that we don't have two dependencies with the same use-case (date/time handling).